itests/gateway_test.go

```diff
@@ -185,52 +190,72 @@ func TestGatewayMsigCLI(t *testing.T) {
 	//stm: @CHAIN_SYNCER_NEW_PEER_HEAD_001, @CHAIN_SYNCER_VALIDATE_MESSAGE_META_001, @CHAIN_SYNCER_STOP_001
 	kit.QuietMiningLogs()
 
-	blocktime := 5 * time.Millisecond
 	ctx := context.Background()
-	nodes := startNodesWithFunds(ctx, t, blocktime, maxLookbackCap, maxStateWaitLookbackLimit)
+	nodes := startNodes(ctx, t, withFunds())
 
 	lite := nodes.lite
 	multisig.RunMultisigTests(t, lite)
 }
 
 type testNodes struct {
-	lite  *kit.TestFullNode
-	full  *kit.TestFullNode
-	miner *kit.TestMiner
+	lite        *kit.TestFullNode
+	full        *kit.TestFullNode
+	miner       *kit.TestMiner
+	gatewayAddr string
+	rpcCloser   jsonrpc.ClientCloser
 }
 
-func startNodesWithFunds(
-	ctx context.Context,
-	t *testing.T,
-	blocktime time.Duration,
-	lookbackCap time.Duration,
-	stateWaitLookbackLimit abi.ChainEpoch,
-) *testNodes {
-	nodes := startNodes(ctx, t, blocktime, lookbackCap, stateWaitLookbackLimit)
+type startOptions struct {
+	blocktime                 time.Duration
+	lookbackCap               time.Duration
+	stateWaitLookbackLimit    abi.ChainEpoch
+	fund                      bool
+	perConnectionAPIRateLimit int
+	perHostRequestsPerMinute  int
+	nodeOpts                  []kit.NodeOpt
+}
 
-	// The full node starts with a wallet
-	fullWalletAddr, err := nodes.full.WalletDefaultAddress(ctx)
-	require.NoError(t, err)
+type startOption func(*startOptions)
 
-	// Get the lite node default wallet address.
-	liteWalletAddr, err := nodes.lite.WalletDefaultAddress(ctx)
-	require.NoError(t, err)
+func applyStartOptions(opts ...startOption) startOptions {
```

Comment:
Subjective: I would name this `newStartOptions` because that's what it does. "apply"ing is an implementation detail that happens as part of construction.